CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Changed
 
 - Jaeger remote sampler's probabilistic strategy now uses the same sampling algorithm as `trace.TraceIDRatioBased` in `go.opentelemetry.io/contrib/samplers/jaegerremote`. (#6892)
+- Remove `request.BodyWrapper` from `r.Body` after the request is processed to avoid issues with handling the `r.Body` by `http/server.go` (which checks the type of `r.Body`) in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` and `go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux`. (#6914)
```

Comment:
Changelog entries should focus on the end-user impact, not internal method names.

```suggestion
- Remove the custom body wrapper from the request's body after the request is processed to allow body type comparisons with the original type `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` and `go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux`. (#6914)
```